crates/bevy_asset/src/asset_server.rs

```diff
@@ -202,8 +205,7 @@ impl AssetServer {
                     LoadState::Loading => {
                         load_state = LoadState::Loading;
                     }
-                    LoadState::Failed => return LoadState::Failed,
-                    LoadState::NotLoaded => return LoadState::NotLoaded,
+                    state => return state,
```

Comment:
I don't think you should replace the clauses for each individual value with a catch-all. This means that the compiler will not exhaustiveness-check the match in the future, if new enum variants are added. This makes the code less maintainable.

I think it's better to explicitly write out each possible enum value, in this scenario.